auth.c

```diff
@@ -53,13 +68,95 @@ void janus_auth_deinit(void) {
 	if(allowed_plugins != NULL)
 		g_hash_table_destroy(allowed_plugins);
 	allowed_plugins = NULL;
+	if(auth_secret != NULL)
+		g_free(auth_secret);
+	auth_secret = NULL;
 	janus_mutex_unlock(&mutex);
 }
 
+gboolean janus_auth_check_signature(const char *token, const char *realm) {
+	if (!auth_enabled || auth_secret == NULL)
+		return FALSE;
+	gchar *token_signature = strchr(token, ':') + 1;
+	gchar *token_metadata = g_strndup(token, token_signature - token - 1);
+	gchar **list = g_strsplit(token_metadata, ",", 3);
+	/* need at least an expiry timestamp and realm */
+	if(!list[0] || !list[1])
+		goto fail;
+	/* verify timestamp */
+	int expiry_time = atoi(list[0]);
+	int real_time = janus_get_real_time() / 1000000;
```

Comment:
`expiry_time` and `real_time` should probably be `int64_t` or `gint64`, not `int`: in 32 bit systems (if they're still around) they won't be large enough to contain the required value. Better to be precise, and in this case stick to the type of the return value of `janus_get_real_time()`. Same considerations for the same thing you do in `janus_auth_check_signature_contains`.